### DISCUSSION TITLE
KeeShare kdbx.share file security

### DISCUSSION THREAD

**Discussion by lwOvOwl** [keepassxreboot/keepassxc#6189] [Q&A] [answered]
Hi everybody, I just checked out KeeShare, and it created a xy.kdbx.share archive that contains a kdbx file and a signature file.
I opened the kdbx like a normal db and I found it has "weak" encryption settings  AES 256 / AES-KDF (KDBX 3.1) / 1kk round, my 1 sec benchmark makes it 65kk. Also the database it originated from use Argon2d. It looks to me Keepassxc decide on its own currently how to protect this shared dbs on creation. I know the chosen encryption not that weak but it still 200 times weaker than the one I use for my database.

I like to ask for more insight into how is the security work here? thanks!

**Comment by droidmonkey** [comment, marked as answer]
It's on my list for the keeshare refactor. Keeshare was third party submitted contribution and needs a lot of work.

**Reply by JD-aus** [reply]
it definately needs alot of work, the basics are great, but understanding how to add keeshares into an existing keepass DB is the key ... i cannot find any information on how to do this anywhere online. the docs on the website do not contain enough information, can the docs be updated, or somebody in the know write a blog post or something.
